### PR TITLE
P4-2434 removing now redundant code and some refactorings to boot.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
Removing now redundant code e.g. password protection of spreadsheet and crude locking mechanism.

### CODE REVIEW

File: helm_deploy/calculate-journey-variable-payments/templates/_envs.tpl [hunk 1]
```diff
@@ -97,11 +97,5 @@ env:
         name: {{ template "app.name" . }}
         key: BASM_AWS_SECRET_ACCESS_KEY
 
-  - name: SPREADSHEET_PASSWORD
```
Comment (matt-crowson-moj) [inline helm_deploy/calculate-journey-variable-payments/templates/_envs.tpl:100]: FYI @mafonso we no longer need this secret.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/config/ImporterConfiguration.kt [hunk 2]
```diff
@@ -26,17 +21,4 @@ class ImporterConfiguration {
 
     @Bean
     fun timeSource() = TimeSource { LocalDateTime.now(Clock.systemDefaultZone()) }
-
-    @Bean
-    fun spreadsheetProtection(@Value("\${SPREADSHEET_PASSWORD:}") value: String?): SpreadsheetProtection {
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/config/ImporterConfiguration.kt:31]: Moved to a separate service.  Does not really belong in here anymore.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/service/ImportService.kt [hunk 3]
```diff
@@ -4,74 +4,51 @@ import org.slf4j.LoggerFactory
...
         private val locationsImporter: LocationsImporter,
         private val priceImporter: PriceImporter,
         private val reportImporter: ReportImporter,
-        private val pricesSpreadsheetGenerator: PricesSpreadsheetGenerator,
         private val reportModelPersister: MoveModelPersister) {
 
     private val logger = LoggerFactory.getLogger(javaClass)
 
-    // Imposing a temporary crude locking solution to prevent DB clashes/conflicts.  I know this will not scale!
-    private val lock = AtomicBoolean(false)
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/service/ImportService.kt:30]: Locking no longer needed.  @mafonso we should no longer be limited to one pod.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/service/ImportService.kt [hunk 4]
```diff
@@ -4,74 +4,51 @@ import org.slf4j.LoggerFactory
...
+    /**
+     * @return a pair representing the return result of the importer, and an import status
+     */
+    private fun <T> import(import: () -> T): T? {
+        logger.info("Attempting import of ${import.javaClass}")
 
-        logger.info("Generating spreadsheet for supplier '$supplierName', moves from '$startDate''")
+        val start = timeSource.dateTime()
 
-        return pricesSpreadsheetGenerator.generate(Supplier.valueOfCaseInsensitive(supplierName), startDate)
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/service/ImportService.kt:75]: Moved to a separate service.  Does not really belong in here anymore.

**Review by jcavell — DISMISSED** [review]

**Review by jcavell — APPROVED** [review]

**Pull request merged**